src/pangolin_viewer/viewer.cc

```diff
@@ -289,14 +289,15 @@ void viewer::draw_landmarks() {
     glBegin(GL_POINTS);
 
     for (const auto lm : landmarks) {
-        glColor3ub(lm->color_[0],lm->color_[1], lm->color_[2]);
-
         if (!lm || lm->will_be_erased()) {
             continue;
         }
         if (*menu_show_local_map_ && local_landmarks.count(lm)) {
             continue;
         }
+
+        glColor3ub(lm->color_[0], lm->color_[1], lm->color_[2]);
```

Comment:
This is just a minor optimization, not really needed but thought might be useful.